presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -775,6 +795,64 @@ private void queryCompletedEvent(Optional<ExecutionFailureInfo> failureInfo)
             queryInfoOutputPath.ifPresent(path -> writeQueryInfo(path, queryInfo, queryInfoJsonCodec));
         }
 
+        private void processShuffleStats()
+        {
+            List<PrestoSparkShuffleStats> statsList = new ArrayList<>(shuffleStatsCollector.value());
+            Map<ShuffleStatsKey, List<PrestoSparkShuffleStats>> statsMap = new HashMap<>();
+            Set<Integer> fragments = new HashSet<>();
+            for (PrestoSparkShuffleStats stats : statsList) {
+                ShuffleStatsKey key = new ShuffleStatsKey(stats.getFragmentId(), stats.getOperation());
+                statsMap.computeIfAbsent(key, (ignored) -> new ArrayList<>()).add(stats);
+                fragments.add(stats.getFragmentId());
+            }
+            List<Integer> sortedFragments = new ArrayList<>(fragments);
+            sortedFragments.sort(null);
+            log.info("Shuffle statistics summary:");
+            for (Integer fragment : sortedFragments) {
+                List<PrestoSparkShuffleStats> readStatistics = statsMap.get(new ShuffleStatsKey(fragment, READ));
+                if (readStatistics != null) {
+                    logShuffleStatsSummary(fragment, READ, readStatistics);
+                }
+                List<PrestoSparkShuffleStats> writeStatistics = statsMap.get(new ShuffleStatsKey(fragment, WRITE));
+                if (writeStatistics != null) {
+                    logShuffleStatsSummary(fragment, WRITE, writeStatistics);
+                }
+            }
+        }
+
+        private void logShuffleStatsSummary(int fragmentId, Operation operation, List<PrestoSparkShuffleStats> statsList)
+        {
+            long totalProcessedRows = 0;
+            long totalProcessedBytes = 0;
+            long totalElapsedWallTimeMills = 0;
+            for (PrestoSparkShuffleStats stats : statsList) {
+                totalProcessedRows += stats.getProcessedRows();
+                totalProcessedBytes += stats.getProcessedBytes();
+                totalElapsedWallTimeMills += stats.getElapsedWallTimeMills();
+            }
+            long totalElapsedWallTimeSeconds = totalElapsedWallTimeMills / 1000;
```

Comment:
hmmm.. if it's just for logging purpose, it can be `double` right? (instead of `long`) , so we don't need to worry about whether `totalElapsedWallTimeSeconds` becomes 0 when the total time is less than 1 second.

Reply:
Then we might either end up with `NaN` or `1_000_000+++ rows/second` that I personally find a little confusing. But I don't have a strong opinion here.